packages/devtools/lib/src/timeline/cpu_profile_protocol.dart

```diff
@@ -89,7 +89,7 @@ class CpuProfileProtocol {
         firstAmpersandIndex < firstPeriodIndex &&
         name.length > firstAmpersandIndex + 1) {
       final nextCharCodeUnit = name[firstAmpersandIndex + 1].codeUnitAt(0);
-      if (isLetter(nextCharCodeUnit)) {
+      if (isLetter(nextCharCodeUnit) || nextCharCodeUnit == '_'.codeUnitAt(0)) {
```

Comment:
why isn't the presence of an ampersand by itself sufficient regardless of what the next character is?
For example, `$MyClass` is a valid but bad Dart class name but I think this pattern wouldn't catch that.
Could make sense to move this to the devtools/lib/src/utils
directory as this is general cleanup for stack trace frame names that we want to apply everywhere including the debugger.

Reply:
Because we have some class names like this:
`dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array...`, which has an ampersand but should not be trimmed.

I suppose we could just check that the following character is not a comma, but that feels riskier to me.